packages/wrangler/src/dev.tsx

```diff
@@ -860,19 +860,12 @@ function useHotkeys(initial: useHotkeysInitialState, port: number) {
   return toggles;
 }
 
-function ErrorFallback(props: {
-  error: Error;
-  resetErrorBoundary: () => void;
-}) {
-  useEffect(() => {
-    console.error(props.error);
-    process.exit(1);
-  });
+function ErrorFallback(props: { error: Error }) {
   return (
-    <Box>
+    <>
       <Text>Something went wrong:</Text>
       <Text>{props.error.message}</Text>
-    </Box>
+    </>
```

Comment:
This change just simplifies this error boundary component:
- the `resetErrorBoundary()` callback was never used
- the `Box` tried to spread the two `Text` elements across the page, which actually looks weird if the terminal is not that wide.
- the `useEffect()` is not ideal because calling `process.exit()` makes it very hard to test errors
- removing the `process.exit()` does mean that the process exits with a zero exit code; but since `dev` is an interactive tool, I don't know if anyone would actually care about the exit code??

Reply:
I'm not sure of that. Does it actually exit without the effect? Or We'll have to actually reset the error boundary on `<some condition>`, because otherwise the user has to manually exit the program and start it up again. This doesn't feel like better UX. 
Either we mock the process.exit call, or do something else to make testing this easier.

Reply:
It does indeed exit without any effect:

```
➜  wrangler git:(dev-tests-1) npm start -- dev --public foo --format service-worker index.ts

> wrangler@0.0.6 start
> npm run bundle && NODE_OPTIONS=--enable-source-maps ./bin/wrangler.js "dev" "--public" "foo" "--format" "service-worker" "index.ts"


> wrangler@0.0.6 bundle
> node -r esbuild-register scripts/bundle.ts

Something went wrong:
You cannot use the service worker format with a `public` directory.
➜  wrangler git:(dev-tests-1) 
```

It might be that when you use `resetErrorBoundary()` then this would prevent the renderer from exiting. But I could not get this to work.

In the meantime this is no worse UX that just running `process.exit()` in the `useEffect()` hook. When we get to the point where we want to add error recovery to the `Dev` component we could add that back in.

I just realised that we can actually set the exit code like this:

```
  useEffect(() => {
    process.exitCode = 1;
  });
```

Reply:
```
> npm run bundle && NODE_OPTIONS=--enable-source-maps ./bin/wrangler.js "dev" "--public" "foo" "--format" "service-worker" "index.ts"


> wrangler@0.0.6 bundle
> node -r esbuild-register scripts/bundle.ts

Something went wrong:
You cannot use the service worker format with a `public` directory.
➜  wrangler git:(dev-tests-1) ✗ echo $?
1
```

Reply:
From looking at the docs, it seems that the correct way to run ink apps, like `Dev`, is render the app: 

```ts
const {waitUntilExit} = render(<Dev />);
await waitUntilExit();
```

And inside the error boundary we can do:

```
const {exit} = useApp();
...
exit() // normal exit
...
exit(error) // abnormal exit
```

Reply:
I updated the PR with a fixup commit that demonstrates this...

Reply:
This looks much better!